src/components/GameBox/GameBox.js

```diff
@@ -0,0 +1,471 @@
+import React, { useEffect, useState } from "react";
+import { BOX_TYPES } from "../../Constants";
+import { showToast, curry } from "../../utilities";
+import { monopolyInstance } from "../../models/Monopoly";
+import { ActionPopup } from "./ActionPopup";
+import "./gameBox.scss";
+
+export const GameBox = (props) => {
+  const {
+    type,
+    name,
+    id,
+    players,
+    currentPlayer,
+    toggleCurrentTurn,
+    baserent,
+    boxType,
+    diceValues,
+    removePlayerFromGame,
+  } = props;
+  const [playerAction, setPlayerAction] = useState(null);
+
+  // const isGo = () => name === "GO";
+
+  const getPropertyOwner = () => {
+    return [...players].find((player) =>
+      player.ownedProperties.find((property) => property.name === name)
+    );
+  };
+
+  const getCurrentRentForProperty = () => {
+    const currentOwner = getPropertyOwner();
+    if (!currentOwner) return 0;
+    const propertyDetails = currentOwner.ownedProperties.find(
+      (property) => property.name === name
+    );
+    if (boxType.type === BOX_TYPES.AVENUE) {
+      if (!propertyDetails.rentLevel) return baserent;
+      else return props[`rent${propertyDetails.rentLevel}`];
+    } else if (boxType.type === BOX_TYPES.RAILROADS) {
+      const allRailRoads = currentOwner.ownedProperties.filter(
+        (property) => property.type === BOX_TYPES.RAILROADS
+      );
+      if (allRailRoads.length === 1) return 25;
```

Comment:
keep the amount 25,50, 75, and 100 in constant variables to increase code readability.